Control startup behavior based on launch configuration

msft-jlange opened the pull request:
Defines a new structure that encapsulates launch configuration into a configuration object and controls the behavior of the launch based on the encapsulated configuration

00xc commented on src/svsm.rs:
```diff
@@ -449,31 +449,43 @@ pub extern "C" fn svsm_main() {
-    let fw_meta = map_and_parse_fw_meta()
-        .unwrap_or_else(|e| panic!("Failed to parse FW SEV meta-data: {:#?}", e));
+    let fw_metadata: Option<SevFWMetaData>;
+    if config.should_launch_fw() {
+        fw_metadata = Some(
+            map_and_parse_fw_meta()
+            .unwrap_or_else(|e| panic!("Failed to parse FW SEV meta-data: {:#?}", e)));
+    } else {
+        fw_metadata = None;
+    }
```
The `fw_metadata` can be constructed directly:

```suggestion
    let fw_metadata = if config.should_launch_fw() {
        Some(
            map_and_parse_fw_meta()
                .unwrap_or_else(|e| panic!("Failed to parse FW SEV meta-data: {:#?}", e)),
        )
    } else {
        None
    };
```

Or even better:

```suggestion
    let fw_metadata = config.should_launch_fw().then(|| {
        map_and_parse_fw_meta()
            .unwrap_or_else(|e| panic!("Failed to parse FW SEV meta-data: {:#?}", e))
    });
```

00xc reviewed (commented):
Overall looks good to me, just left one suggestion below. Also, please sign your commits via `git commit -s` and run your changes through `cargo fmt` and `cargo clippy`. CI should have warned you about these but it's disabled because it's your first contribution.

msft-jlange commented:
Thanks for the feedback!  I’m going to go with the first of your two proposed options, since the essence of any good code is readability, and the second option uses esoteric Rust syntax that I personally find much less readable (plus the second option will be much more difficult to integrate with future changes that integrate a different option provider).

-Jon

From: Carlos López ***@***.***>
Sent: Wednesday, November 29, 2023 1:13 AM
To: coconut-svsm/svsm ***@***.***>
Cc: Jon Lange ***@***.***>; Author ***@***.***>
Subject: Re: [coconut-svsm/svsm] Control startup behavior based on launch configuration (PR #168)


@00xc commented on this pull request.

Overall looks good to me, just left one suggestion below. Also, please sign your commits via git commit -s and run your changes through cargo fmt and cargo clippy. CI should have warned you about these but it's disabled because it's your first contribution.

________________________________

In src/svsm.rs<https://github.com/coconut-svsm/svsm/pull/168#discussion_r1408962193>:

> +    if config.should_launch_fw() {

+        fw_metadata = Some(

+            map_and_parse_fw_meta()

+            .unwrap_or_else(|e| panic!("Failed to parse FW SEV meta-data: {:#?}", e)));

+    } else {

+        fw_metadata = None;

+    }

The fw_metadata can be constructed directly:
⬇️ Suggested change

-    if config.should_launch_fw() {

-        fw_metadata = Some(

-            map_and_parse_fw_meta()

-            .unwrap_or_else(|e| panic!("Failed to parse FW SEV meta-data: {:#?}", e)));

-    } else {

-        fw_metadata = None;

-    }

+    let fw_metadata = if config.should_launch_fw() {

+        Some(

+            map_and_parse_fw_meta()

+                .unwrap_or_else(|e| panic!("Failed to parse FW SEV meta-data: {:#?}", e)),

+        )

+    } else {

+        None

+    };

Or even better:
⬇️ Suggested change

-    if config.should_launch_fw() {

-        fw_metadata = Some(

-            map_and_parse_fw_meta()

-            .unwrap_or_else(|e| panic!("Failed to parse FW SEV meta-data: {:#?}", e)));

-    } else {

-        fw_metadata = None;

-    }

+    let fw_metadata = config.should_launch_fw().then(|| {

+        map_and_parse_fw_meta()

+            .unwrap_or_else(|e| panic!("Failed to parse FW SEV meta-data: {:#?}", e))

+    });

—
Reply to this email directly, view it on GitHub<https://github.com/coconut-svsm/svsm/pull/168#pullrequestreview-1754755854>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AJKNAYUMIDBBLCFQKMZRNQ3YG34A5AVCNFSM6AAAAAA764N4W6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTONJUG42TKOBVGQ>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

00xc reviewed (commented):
I'm still seeing some clippy warnings (`cargo clippy`). Would you mind fixing those as well? Thanks!

msft-jlange commented:
Strange - I did run cargo clippy and it didn't give me any errors.  And now that you've prompted me to run it again, it did generate warnings.  I've fixed them and updated the PR.  Thanks for being patient with me!

-Jon

From: Carlos López ***@***.***>
Sent: Wednesday, November 29, 2023 9:58 AM
To: coconut-svsm/svsm ***@***.***>
Cc: Jon Lange ***@***.***>; Author ***@***.***>
Subject: Re: [coconut-svsm/svsm] Control startup behavior based on launch configuration (PR #168)


@00xc commented on this pull request.

I'm still seeing some clippy warnings (cargo clippy). Would you mind fixing those as well? Thanks!

-
Reply to this email directly, view it on GitHub<https://github.com/coconut-svsm/svsm/pull/168#pullrequestreview-1755897681>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AJKNAYVXEVCEL7WFDOHX72DYG5ZRFAVCNFSM6AAAAAA764N4W6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTONJVHA4TONRYGE>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>
